### ISSUE TITLE
Problems installing py3status

### ISSUE THREAD

**Issue by gerritwalther** [ultrabug/py3status#61]
Hi,

I've got some problems installing the newest version of py3status. I don't think its related to the new version, more some issues with my system, but maybe you can help me. I'm on Debian testing.

I installed py3status (v2.1) about 9 days ago via pip:
```
sudo pip install py3status
```
Used sudo because pip wanted to write it to `/usr/local/bin`. Everything worked fine.

Today I wanted to adjust the output of the battery_level module. Nothing changed and I tried to upgrade to the newest version. Here is where I'm getting problems.

I upgraded using pip:
```
sudo pip install --upgrade py3status
```
It upgraded well. After that the permission of file `/usr/local/bin/py3status` was set to 700. Which prevented i3 to load it. I changed it to `755` in hope to change the problems. But when I try to check the version as a normal user I get the following stacktrace:
```
Traceback (most recent call last):
  File "/usr/local/bin/py3status", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2876, in <module>
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 449, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 745, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 639, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: py3status==2.2
```
After some googling I found out that this might mean that there are multiple versions installed, but neither `pip freeze` nor `sudo pip freeze` list multiple versions.
```
$ sudo pip freeze
argparse==1.2.1
cffi==0.8.6
chardet==2.3.0
colorama==0.3.2
cryptography==0.6.1
html5lib==0.999
ndg-httpsclient==0.3.2
numpy==1.8.2
ply==3.4
py3status==2.2
pyOpenSSL==0.14
pyasn1==0.1.7
pycparser==2.10
pygobject==3.14.0
requests==2.4.3
six==1.8.0
urllib3==1.9.1
wheel==0.24.0
wsgiref==0.1.2
```
```
$ pip freeze
argparse==1.2.1
cffi==0.8.6
chardet==2.3.0
colorama==0.3.2
cryptography==0.6.1
html5lib==0.999
ndg-httpsclient==0.3.2
numpy==1.8.2
ply==3.4
pyOpenSSL==0.14
pyasn1==0.1.7
pycparser==2.10
pygobject==3.14.0
requests==2.4.3
six==1.8.0
urllib3==1.9.1
wheel==0.24.0
wsgiref==0.1.2
```

I tried to install version 2.1 with pip, but that didn't change anything and I removed it. I hope you can give me any hint on how to solve this. I'm already missing using py3status ;)

**Comment by ultrabug** [comment]
Hi @gerritwalther, you're actually the second person to report this problem on Debian.

I'll have to install a VM myself to try and debug this weird issue which is permission related.
- Can you give me your exact Debian version plz ?

The workaround for the other user was to uninstall py3status using `sudo pip uninstall py3status` then install it again still using `sudo pip install py3status` and fix the permissions on the `py3status` executable using `sudo usermod -a -G staff YOURUSER` and then relog your session.

Basically this adds your current user to the `staff` group which has the right to execute py3status as you should be able to see by issuing `sudo ls -l /usr/local/bin/py3status`.

Now why does Debian do this I have no clue yet :(

**Comment by gerritwalther** [comment]
I'm using the Debian testing branch.
```
$ uname -a
Linux SLPN-NB-GWA 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt2-1 (2014-12-08) x86_64 GNU/Linux
```
```
$ ls -al /usr/local/bin/py3status 
-rwx------ 1 root staff 305 Jan 28 20:08 /usr/local/bin/py3status
```
Will  try the workaround now and post here, if its working.

**Comment by gerritwalther** [comment]
I've set the permission level now to `755` and now get a different error than before:
```
$ py3status -v
Traceback (most recent call last):
  File "/usr/local/bin/py3status", line 9, in <module>
    load_entry_point('py3status==2.2', 'console_scripts', 'py3status')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 356, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2475, in load_entry_point
    raise ImportError("Entry point %r not found" % ((group, name),))
ImportError: Entry point ('console_scripts', 'py3status') not found
```

Digged a little further. The folders installed in `/usr/local/lib/python2.7/dist-packages` had set the `S` permission. Have removed that bit, also set the folders to `755` and files to `644`, now everything is working ^^

For reference:
Before:
```
/usr/local/lib/python2.7/dist-packages $ ll
total 20K
drwxrwsr-x 4 root staff 4.0K Jan 28 21:13 .
drwxrwsr-x 4 root staff 4.0K Jan 28 15:02 ..
-rw------- 1 root staff  185 Jan 28 20:05 easy-install.pth
drwx--S--- 3 root staff 4.0K Jan 28 21:13 py3status
drwx--S--- 2 root staff 4.0K Jan 28 21:13 py3status-2.2.egg-info


/usr/local/lib/python2.7/dist-packages $ sudo ls -al py3status
total 108
drwx------ 3 root staff  4096 Jan 28 21:13 .
drwxrwsr-x 4 root staff  4096 Jan 28 21:13 ..
-rw------- 1 root staff 56459 Jan 28 21:13 __init__.py
-rw------- 1 root staff 39943 Jan 28 21:13 __init__.pyc
drwx--S--- 2 root staff  4096 Jan 28 21:13 modules
```
Afterwards:
```
/usr/local/lib/python2.7/dist-packages $ ll
total 20K
drwxrwsr-x 4 root staff 4.0K Jan 28 21:13 .
drwxrwsr-x 4 root staff 4.0K Jan 28 15:02 ..
-rw------- 1 root staff  185 Jan 28 20:05 easy-install.pth
drwxr-xr-x 3 root staff 4.0K Jan 28 21:13 py3status
drwxr-xr-x 2 root staff 4.0K Jan 28 21:13 py3status-2.2.egg-info


/usr/local/lib/python2.7/dist-packages $ ll py3status
total 108K
drwxr-xr-x 3 root staff 4.0K Jan 28 21:13 .
drwxrwsr-x 4 root staff 4.0K Jan 28 21:13 ..
-rw-r--r-- 1 root staff  56K Jan 28 21:13 __init__.py
-rw-r--r-- 1 root staff  40K Jan 28 21:13 __init__.pyc
drwxr-xr-x 2 root staff 4.0K Jan 28 21:13 modules
```

Tell me if you need further help solving the problem. Or just for testing.

**Comment by gerritwalther** [comment]
Another interesting thing I found out today while installing i3 and py3status on my desktop machine:

I had no problems ^^ The permissions were correct (just as I changed them in the last post). So  the main problem seems to be with reinstalling py3status. Maybe its even a problem with pip or easy_install.

My desktop has Debian testing as well, my user is **not** in the staff group.

**Comment by ultrabug** [comment]
I've not forgotten this issue, please bear with me I'm quite busy lately.

**Comment by gerritwalther** [comment]
Totally fine. I've got everything working, so there's no need to hurry.

**Comment by gerritwalther** [comment]
Oh man, I think this is really a pip or debian problem. I installed evdev for python today via pip and its the same permission problem as it is with py3status.

I will try to find a solution for this or file a bug for pip.

**Comment by FedericoCeratto** [comment]
@gerritwalther: it's never recommended to use pip install on Debian (or any package-based distribution). On Debian you can use pypi-install

**Comment by ultrabug** [comment]
@FedericoCeratto and do you know if using pypi-install solves this issue by any chance ?

**Comment by FedericoCeratto** [comment]
@ultrabug: I didn't try to replicate the issue; Anyhow, sudo pypi-install py3status worked just fine as today and installed version 2.3

**Comment by gerritwalther** [comment]
@FedericoCeratto Thank you for the hint. Had to remove py3status with `sudo pip uninstall py3status` prior to using pypi-install. But now its working fine.

For everyone else: pypi-install is in the python-stdeb package.

**Comment by ultrabug** [comment]
I added a note about this in the README, thanks guys.

This is while the debian package by @FedericoCeratto is coming live to apt ;)
